client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPushConsumerImpl.java

```diff
@@ -280,6 +280,12 @@ public void onSuccess(PullResult pullResult) {
 
                     switch (pullResult.getPullStatus()) {
                         case FOUND:
+                            if (pullResult.getMsgFoundList().size() == 0) {
```

Comment:
It maybe confusing without comments.
For if the status is FOUND, it is expected that the msgList size will be greater than 0.